src/libraries/System.Private.CoreLib/src/System/Int64.cs

```diff
@@ -319,23 +319,41 @@ object IConvertible.ToType(Type type, IFormatProvider? provider)
         public static long TrailingZeroCount(long value) => BitOperations.TrailingZeroCount(value);
 
         /// <inheritdoc cref="IBinaryInteger{TSelf}.GetShortestBitLength()" />
-        long IBinaryInteger<long>.GetShortestBitLength()
+        int IBinaryInteger<long>.GetShortestBitLength()
         {
             long value = m_value;
 
             if (value >= 0)
             {
-                return (sizeof(long) * 8) - LeadingZeroCount(value);
+                return (sizeof(long) * 8) - BitOperations.LeadingZeroCount((ulong)(value));
```

Comment:
Just double checking that this is actually the semantic that we want for negative numbers.

The positive number 1 will report 1 bit, but a negative 1 will report 64 bits, even though any of 1, 8(, 16, 24, 32, ...), 64 are valid.  i.e. for ASN.1/DER INTEGER values -1 has to be encoded always as the single-byte value `{ 0xFF }`.

Reply:
Nevermind, I missed the if.